File: yh_dual/scripts/yh_dual_pub.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import rospy
from yh_dual.msg import YhDual

def yh_dual_pub():
    rospy.init_node("yh_dual_pub")
    pub = rospy.Publisher("yh_dual_topic", YhDual, queue_size=10)
    loop_rate = rospy.Rate(8)
    msg = YhDual()

    while not rospy.is_shutdown():
        msg.stamp = rospy.Time.now()
        pub.publish(msg)
        msg.data += 1
        loop_rate.sleep()

if __name__ == "__main__":
    try:
        yh_dual_pub()
    except rospy.ROSInterruptException:
        pass

File: yh_dual/src/yh_dual_pub.cpp
#include "ros/ros.h"
#include "yh_dual/YhDual.h"

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_dual_pub");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<yh_dual::YhDual>("yh_dual_topic", 100);

    ros::Rate loop_rate(8);
    yh_dual::YhDual msg;

    while (ros::ok())
    {
        msg.stamp = ros::Time::now();
        pub.publish(msg);
        msg.data++;
        loop_rate.sleep();
    }

    return 0;
}